src/puffin/src/file_format/writer.rs

```diff
@@ -30,7 +31,10 @@ pub struct Blob<R> {
     pub blob_type: String,
 
     /// The data of the blob
-    pub data: R,
+    pub compressed_data: R,
+
+    /// The codec used to compress the blob.
+    pub compression_codec: Option<CompressionCodec>,
```

Comment:
How about define `None` also as a "uncompressed" branch of `CompressionCodec`, like `CompressionType`

Reply:
I'm quite happy now. It doesn't bring much profit but there are certain modification costs.